archive_upload/handlers/dsmc_handlers.py

```diff
@@ -0,0 +1,754 @@
+            raise ArteriaUsageException("Error when getting description from PDC. dsmc returned != 0. Output:".format(dsmc_out))
+
+        log.debug("Raw output from dsmc: {}".format(dsmc_out))
+
+        uploaded_versions = [line.strip() for line in dsmc_out if path_to_archive in line]
+
+        if not uploaded_versions:
+            raise ArteriaUsageException("Error when getting description from PDC. No descriptions available for {}".format(path_to_archive))
+
+        log.debug("Found the following uploaded versions of this archive: {}".format(uploaded_versions))
+
+        # Uploads are chronologically sorted, with the latest upload last.
+        latest_upload = uploaded_versions[-1:][0]
+
+        # We need the description of this upload: the last field. E.g.:
+        # 4,096  B  01/10/2017 16:47:24    /data/mm-xart002/runfolders/johanhe_test_0809_001-AG2UJ_archive Never a33623ba-55ad-4034-9222-dae8801aa65e
+        latest_descr = latest_upload.split()[-1:][0]
+        log.debug("Latest uploaded version is {} with description {}".format(latest_upload, latest_descr))
+
+        return latest_descr
+
+    def get_pdc_filelist(self, path_to_archive, descr, dsmc_log_dir):
+        """
+        Gets the files and their sizes from PDC for a certain path (archive), with a specific description.
+
+        :param path_to_archive: The path to the archive
+        :param descr: The description label for the uploaded archive
+        :return The dict `uploaded_files` containing a mapping between uploaded file and size in bytes
+        """
+        log.info("Fetching remote filelist for {} from PDC...".format(path_to_archive))
+        cmd = "export DSM_LOG={} && dsmc q ar {}/ -subdir=yes -description={}".format(dsmc_log_dir, path_to_archive, descr)
+
+        p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+
+        dsmc_out, _ = p.communicate()
+        dsmc_out = dsmc_out.splitlines()
+
+        if p.returncode != 0:
+            raise ArteriaUsageException("Error when getting filelist from PDC. Output:".format(dsmc_out))
+
+        # We're only interested in the lines from the dsmc output that contains the
+        # path to the archive.
+        matched_lines = [line.strip() for line in dsmc_out if path_to_archive in line]
+
+        if not matched_lines:
+            raise ArteriaUsageException("Error when getting filelist from PDC. No files uploaded for {}".format(path_to_archive))
+
+        log.debug("Uploaded files to PDC: {}".format(matched_lines))
+
+        uploaded_files = {}
+
+        # We need to convert the sizes to a common format for easier comparison with local size.
+        # An output line can look like
+        #4,096  B  2017-07-27 17.48.34    /data/mm-xart002/runfolders/johanhe_test_0809_001-AG2UJ_archive/Config Never e374bd6b-ab36-4f41-94d3-f4eaea9f30d4
+        # but varies, depending on the environment's locale. Can e.g. be "4 096",
+        for line in matched_lines:
```

Comment:
I think that this part could be broken out into a separate method, go make the `get_pdc_filelist` shorter and easier to read.